basiclibrary/src/main/java/inheritance/Shop.java

```diff
@@ -0,0 +1,82 @@
+package inheritance;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+public class Shop {
+    private String name;
+    private String description;
+    private int dollarSigns;
+    private int totalStars;
+    private LinkedList reviewList = new LinkedList();
+
+    public Shop(String name, String description, int dollarSigns){
+        this.name = name;
+        this.description = description;
+        this.dollarSigns = dollarSigns;
+        this.totalStars = 0;
+    }
+
+    public String getName(){
+        return this.name;
+    }
+
+    public String getDescription(){
+        return this.description;
+    }
+
+    public int getDollarSigns(){
+        return this.dollarSigns;
+    }
+
+    public void addReview(Review r){
+        if(this.reviewList == null){
+            this.reviewList.add(r);
+        }
+        else if(!checkList(r.toString() ) ){
+            this.reviewList.add(r);
+            this.totalStars += r.stars;
+        }
+        else{
+            throw new IllegalArgumentException("review already exists!");
+        }
+    }
+
+    public boolean checkList(String review){
+        if(this.reviewList == null){
```

Comment:
Because reviewList is instantiated upon creation, I'm not sure it will ever `==` null.
I just tested this by creating an empty ArrayList and it wasn't equal to null:
`import java.util.*;

public class MyClass {
    public static void main(String args[]) {
        ArrayList<Integer> a = new ArrayList<>();

        System.out.println(a == null);
    }
}`